packages/core/src/config/vendure-config.ts

```diff
@@ -411,6 +412,17 @@ export interface ShippingOptions {
     shippingCalculators?: Array<ShippingCalculator<any>>;
 }
 
+export interface FulfillmentOptions {
+    /**
+     * @description
+     * Allows the definition of custom states and transition logic for the fulfillment process state machine.
+     * Takes an array of objects implementing the {@link CustomFulfillmentProcess} interface.
+     *
+     * @default []
+     */
+    process?: Array<CustomFulfillmentProcess<any>>;
```

Comment:
I'm not sure whether we should add a new top-level options object here (in general I want to keep the top-level as small as possible and group related options). We already have `shippingOptions`, and fulfillment is very much related to shipping, so maybe it should live there? I'm not totally sure on this. It depends whether we also foresee other specifically fulfillment-related config being needed in future.

Reply:
Yes, it's a good point. For now I think that we can put that together with Shipping as you said. If someday the Fulfillment grow we can change that.